lib/src/features/city_details/data/data_sources/remote_data_source.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:emotion/src/features/city_details/data/models/city_details_model.dart';
+import 'package:http/http.dart';
+
+abstract class CityDetailsRemoteDataSource {
+  Future<CityDetailsModel> getCityDetails(String cityName, String countryName);
+}
+
+class CityDetailsRemoteDataSourceImpl extends CityDetailsRemoteDataSource {
+  final http.Client client;
+
+  CityDetailsRemoteDataSourceImpl({required this.client});
+
+  @override
+  Future<CityDetailsModel> getCityDetails(String cityName, String countryName) async {
+    final response = await client.get(
+      Uri.parse('https://dev.escooters.blumil.pl/api/$countryName/$cityName')
```

Comment:
Maybe this is the reason why http client is not working?
```suggestion
      Uri.parse('https://dev.escooters.blumilk.pl/api/$countryName/$cityName')
```

Reply:
Fixed typo, thx. The problem isn't there tho. 
<img width="831" alt="image" src="https://github.com/blumilksoftware/e-motion-mobile/assets/102246039/6713f7dd-18bf-46ce-99d7-5d6248191816">
Importing http isn't somewhat working.

Reply:
The problem might be caused by a naming conflict. I noticed that you already used the alias "as http." Did that resolve the issue?

Reply:
yes, using alias did resolve the issue, thanks.